tools/gfx-unit-test/shader-cache-tests.cpp

```diff
@@ -0,0 +1,623 @@
+#include "tools/unit-test/slang-unit-test.h"
+
+#include "slang-gfx.h"
+#include "gfx-test-util.h"
+#include "tools/gfx-util/shader-cursor.h"
+#include "source/core/slang-basic.h"
+
+#include "source/core/slang-memory-file-system.h"
+#include "source/core/slang-file-system.h"
+
+using namespace gfx;
+
+namespace gfx_test
+{
+
+    struct BaseShaderCacheTest
+    {
+        UnitTestContext* context;
+        Slang::RenderApiFlag::Enum api;
+
+        ComPtr<IDevice> device;
+        ComPtr<IPipelineState> pipelineState;
+        ComPtr<IResourceView> bufferView;
+
+        // The test shader needs to physically exist on disk for loadComputeProgram to
+        // correctly load it while the shader cache should exist in memory to avoid needing
+        // to clean up old cache files from previous test runs.
+        ComPtr<ISlangMutableFileSystem> diskFileSystem;
+        ComPtr<ISlangMutableFileSystem> cacheFileSystem;
```

Comment:
I'm a bit confused why the `cacheFileSystem` cannot be used for loading the actual shaders as well, instead of just for storing the cache.

Reply:
As I brought up in the gfx channel last week, using `cacheFileSystem` (which is a `MemoryFileSystem` and not a `RelativeFileSystem` like `diskFileSystem` is) to save the shaders results in `loadComputeProgram()` being unable to find them. I was told the shaders need to be saved to disk in order for `loadComputeProgram()` to find them.

Reply:
I've edited this comment to better reflect that both file systems are needed due to needing to reset the cache between consecutive test runs and to save shaders to disk because that's where `loadComputeProgram()` looks when loading a shader.